compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -559,6 +559,21 @@ pub fn provide(providers: &mut ty::query::Providers) {
         vtable_methods,
         type_implements_trait,
         subst_and_check_impossible_predicates,
+        mir_abstract_const: |tcx, def_id| {
+            let def_id = def_id.as_local()?; // We do not store failed AbstractConst's.
```

Comment:
how does that work? Is this a setting on the query?

Reply:
it will be, rn we don't encode them at all

Reply:
Metadata will have to contain an `Option` anyway, you can't just "not have" anything in metadata, so we can populate the query when reading it from metadata